platforms/platforms.go

```diff
@@ -127,15 +127,15 @@ type Matcher interface {
 	Match(platform specs.Platform) bool
 }
 
-// NewMatcher returns a simple matcher based on the provided platform
```

Comment:
Best to have the exported function in the non-platform specific file then call into non-exported implementations. This will also need an other implementation besides just Linux and Windows

Reply:
+1

Reply:
> This will also need an other implementation besides just Linux and Windows

I've updated to include other implementations, it uses the same defaults as previous except Windows.

> Best to have the exported function in the non-platform specific file then call into non-exported implementations

I don't think I fully understood what I should do here.  could you explain a bit more? Thanks!

Reply:
`NewMatcher` is an exported function. So it should be in platforms.go which calls the un-exported `newDefaultMatcher` in the various platforms_<os>.go files. But actually thats another point. Why is `NewDefaultMatcher` exported? There are no callers right?

Reply:
Thanks, I see what I did and I've updated